presto-main/src/main/java/io/prestosql/operator/scalar/BitwiseFunctions.java

```diff
@@ -75,4 +75,123 @@ public static long bitwiseXor(@SqlType(StandardTypes.BIGINT) long left, @SqlType
     {
         return left ^ right;
     }
+
+    @Description("bitwise left shift")
+    @ScalarFunction("bitwise_left_shift")
+    @SqlType(StandardTypes.TINYINT)
+    public static long bitwiseLeftShiftTinyint(@SqlType(StandardTypes.TINYINT) long num, @SqlType(StandardTypes.INTEGER) long shift)
+    {
+        long shifted = (num << shift);
+        return preserveSign(shifted, 8);
+    }
+
+    @Description("bitwise left shift")
+    @ScalarFunction("bitwise_left_shift")
+    @SqlType(StandardTypes.SMALLINT)
+    public static long bitwiseLeftShiftSmallint(@SqlType(StandardTypes.SMALLINT) long num, @SqlType(StandardTypes.INTEGER) long shift)
+    {
+        long shifted = (num << shift);
+        return preserveSign(shifted, 16);
+    }
+
+    @Description("bitwise left shift")
+    @ScalarFunction("bitwise_left_shift")
+    @SqlType(StandardTypes.INTEGER)
+    public static long bitwiseLeftShiftInteger(@SqlType(StandardTypes.INTEGER) long num, @SqlType(StandardTypes.INTEGER) long shift)
+    {
+        long shifted = (num << shift);
+        return preserveSign(shifted, 32);
+    }
+
+    @Description("bitwise left shift")
+    @ScalarFunction("bitwise_left_shift")
+    @SqlType(StandardTypes.BIGINT)
+    public static long bitwiseLeftShiftBigint(@SqlType(StandardTypes.BIGINT) long num, @SqlType(StandardTypes.INTEGER) long shift)
+    {
+        return num << shift;
+    }
+
+    private static long preserveSign(long shiftedValue, long size)
+    {
+        long mask = (1L << size) - 1;
+        long signedBit = 1L << (size - 1);
+        if ((shiftedValue & signedBit) != 0) {
+            // Preserve the sign in 2's complement format
+            mask = ~0 & ~mask;
```

Comment:
This could also be a constant for each type

Reply:
`~0` = `0b1111111...1111`, so it's a no-op when `&` with ~mask. Thus, this is just `mask = ~mask`. Then, this block could simplify to:

```
if ((shiftedValue & signBit) != 0) {
    return shiftedValue | ~mask;
}

return shiftedValue & mask;
```